app/features/contacts/contact-repository.ts

```diff
@@ -0,0 +1,160 @@
+  username: string;
+};
+
+export class ContactRepository {
+  constructor(private readonly db: BoardwalkDb) {}
+
+  async get(contactId: string) {
+    const query = this.db.from('contacts').select().eq('id', contactId);
+
+    const { data, error } = await query.single();
+
+    if (error) {
+      throw new Error('Failed to get contact', error);
+    }
+
+    return ContactRepository.toContact(data);
+  }
+
+  /**
+   * Gets all contacts for a user
+   * @param ownerId - The ID of the user whose contacts to fetch
+   * @returns Array of contacts
+   */
+  async getAll(
+    ownerId: string,
+    options?: { abortSignal?: AbortSignal },
+  ): Promise<Contact[]> {
+    const query = this.db
+      .from('contacts')
+      .select()
+      .eq('owner_id', ownerId)
+      .order('username', { ascending: true }); // sort alphabetically
+
+    if (options?.abortSignal) {
+      query.abortSignal(options.abortSignal);
+    }
+
+    const { data, error } = await query;
+
+    if (error) {
+      throw new Error('Failed to get contacts', error);
+    }
+
+    return data.map(ContactRepository.toContact);
+  }
+
+  /**
+   * Creates a new contact for a user
+   * @param ownerId - The ID of the user who owns the contact
+   * @param contact - The contact data to create
+   * @returns The created contact
+   */
+  async create(
+    contact: CreateContact,
+    options?: { abortSignal?: AbortSignal },
+  ): Promise<Contact> {
+    const query = this.db
+      .from('contacts')
+      .insert({
+        owner_id: contact.ownerId,
+        username: contact.username,
+      })
+      .select();
+
+    if (options?.abortSignal) {
+      query.abortSignal(options.abortSignal);
+    }
+
+    const { data, error } = await query.single();
+
+    if (error) {
+      throw new Error('Failed to create contact', error);
+    }
+
+    return ContactRepository.toContact(data);
+  }
+
+  /**
+   * Deletes a contact
+   * @param contactId - The ID of the contact to delete
+   */
+  async delete(
+    contactId: string,
+    options?: { abortSignal?: AbortSignal },
+  ): Promise<void> {
+    const query = this.db.from('contacts').delete().eq('id', contactId);
+
+    if (options?.abortSignal) {
+      query.abortSignal(options.abortSignal);
+    }
+
+    const { error } = await query;
+
+    if (error) {
+      throw new Error('Failed to delete contact', error);
+    }
+  }
+
+  /**
+   * Search for user profiles by partial username
+   * @param query - The partial username to search for
+   * @param currentUserId - The ID of the current user
+   * @returns Array of user profiles
+   */
+  async searchUserProfiles(
+    query: string,
+    currentUserId: string,
+    options?: { abortSignal?: AbortSignal; sort?: 'desc' | 'asc' },
+  ): Promise<UserProfile[]> {
+    const trimmedQuery = query.trim();
+    if (trimmedQuery.length < 3) {
+      return [];
+    }
+
+    const rpcQuery = this.db.rpc('find_user_profiles_by_partial_username', {
```

Comment:
I don't think the name of this function works anymore. What function does is it returns the potential contacts so something like `find_non_contact_users` or similar would make more sense imo